test/jobs/maintenance_tasks/task_job_test.rb

```diff
@@ -64,6 +64,22 @@ class TaskJobTest < ActiveJob::TestCase
       assert_no_enqueued_jobs
     end
 
+    test ".perform_now skips iterations when Run is cancelled even when a race condition happens" do
+      run = @run
+      CustomTaskJob.race_condition_hook = -> do
+        Run.find(run.id).update(status: :cancelling)
```

Comment:
Do we need to assign `run = @run`? Can't we just use 
```ruby
CustomTaskJob.race_condition_hook = -> do
  Run.find(@run.id).update(status: :cancelling)
end
```
?

Reply:
Good catch, I needed that before because I used `before_perform` directly.